specs/permissions-schema.json

```diff
@@ -0,0 +1,142 @@
+{
+    "$schema": "https://json-schema.org/draft-07/schema#",
+    "title": "Schema for OAuth Permissions",
+    "type": "object",
+    "properties": {
+        "additionalProperties": false,
+        "permissions": {
+            "title": "Map of permission name to definition",
+            "additionalProperties": false,
+            "type": "object",
+            "patternProperties": {
+                "[\\w]+\\.[\\w]+[\\.[\\w]+]?": {
+                    "$ref": "#/definitions/permission"
+                }
+            }
+        }
+    },
+    "definitions": {
+        "permission": {
+            "type": "object",
+            "title": "Permission definition",
+            "additionalProperties": false,
+            "properties": {
+                "note": {
+                    "type": "string"
+                },
+
+                "implicit": {
+                    "type": "boolean"
+                },
+
+                "provisioningInfo": {
+                    "$ref": "#/definitions/provisioningInfo"
+                },
+                
+                "schemes": {
+                    "type": "object",
+                    "patternProperties": {
+                        ".*": {
+                            "$ref": "#/definitions/scheme"
+                        }
+                    }
+                },
+                "pathSets": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/pathSet"
+                    }
+                }
+            }
+        },
+        "ownerInfo":{
+            "type": "object",
+            "properties": {
+                "ownerSecurityGroup": {
+                    "type": "string"
+                }
+            }
+        },
+        "pathSet": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "schemeKeys": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
```

Comment:
Perhaps we could also list the valid values for the `schemeKeys` to aid in validation.

Reply:
I did that in this PR https://github.com/microsoftgraph/kibali/pull/41